src/UI/Button.elm

```diff
@@ -32,283 +31,502 @@ import Element.Border as Border
 import Element.Events as Events
 import Element.Font as Font
 import UI.Icon as Icon exposing (Icon)
-import UI.Internal.Basics exposing (lazyMap, pairUncurry)
+import UI.Internal.Basics exposing (lazyMap, maybeToList, pairUncurry, prependMaybe)
 import UI.Internal.Palette as Palette
 import UI.Internal.Primitives as Primitives
 import UI.Internal.Size as Size exposing (Size)
 import UI.Internal.Text as Text exposing (TextColor)
 import UI.Link as Link exposing (Link)
-import UI.Palette as Palette exposing (brightnessDarkest, brightnessLight, brightnessLighter, brightnessLightest, brightnessMiddle)
+import UI.Palette as Palette exposing (brightnessDarkest, brightnessLight, brightnessLighter, brightnessLightest, brightnessMiddle, tonePrimary)
 import UI.RenderConfig exposing (RenderConfig)
 import UI.Text as Text
 import UI.Utils.ARIA as ARIA
 import UI.Utils.Element as Element
 
 
 type alias Options =
-    { mode : ButtonMode
-    , tone : ButtonTone
-    , width : ButtonWidth
+    { width : ButtonWidth
     , size : Size
     }
 
 
 type alias Properties msg =
     { body : ButtonBody
-    , click : ButtonClick msg
+    , mode : ButtonMode msg
     }
 
 
 type Button msg
     = Button (Properties msg) Options
+    | Toggle (ToggleProperties msg) Options
 
 
 type ButtonBody
     = BodyText String
     | BodyIcon Icon
 
 
-type ButtonClick msg
-    = ClickMsg msg
-    | ClickLink Link
-    | ClickToggle (Bool -> msg) Bool
+type ButtonAction msg
+    = ActionMsg msg
+    | ActionRedirect Link
 
 
-type ButtonMode
-    = ModeEnabled
-    | ModeDisabled
+type ButtonMode msg
+    = ButtonWorking (ButtonAction msg) ButtonStyle
```

Comment:
Maybe `ButtonActive`? Because in my opinion,`ButtonWorking` seems that the button is doing something in that particular moment, but in fact, it is like waiting for some external action, like `idle`